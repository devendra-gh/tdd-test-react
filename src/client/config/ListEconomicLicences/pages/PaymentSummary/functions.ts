import { IVariables } from '@tamm/app-composer';
import { getDateFromTimeStamp } from 'client/config/utils/datetime';
import {
  getFeeDetails,
  proceedToPayment,
  getPaymentLink,
  addAnalytics,
  PAY1_EVENT_KEY,
} from '../../utils';
import { PATH_HOME } from '../../routes';
import { errorBoundary } from '../../services';

export const startPayment = async (props: IVariables, shouldStart: boolean) => {
  // open paymentLnk in new window
  if (shouldStart) {
    props.actions.showErrors.update(true);
    const unchecked = props.termsAndConditionsValues.includes(false);
    if (!unchecked) {
      props.actions.startingPayment.update(true);
      addAnalytics(
        PAY1_EVENT_KEY,
        { applicationStatus: 'start Payment', serviceStatus: 'success' },
        { sum: props.totalSection },
      );
      window.location.href = props.paymentLink;
      proceedToPayment(props, shouldStart);
    }
  } else {
    await proceedToPayment(props, shouldStart);
    props.history.push(PATH_HOME);
  }
};

export const handleTermsConditions = (props: IVariables, item: any) => {
  const { termsAndConditionsValues, actions } = props;
  const updatedTermsAndConditions = [...termsAndConditionsValues];
  updatedTermsAndConditions[item] = !termsAndConditionsValues[item];
  actions.termsAndConditionsValues.update(updatedTermsAndConditions);
};

export const onPageInit = (props: IVariables) => {
  const tags = [
    {
      label: 'notice.refNo',
      value: props.cnNumber,
    },
    {
      label: 'notice.submit',
      value: getDateFromTimeStamp(props.submitDate),
    },
  ];

  return {
    tags,
  };
};

export const init = async (props: IVariables) => {
  const { i18n, locale } = props;
  let fees = [];
  let paymentLink: any;
  props.actions.showErrors.update(false);
  try {
    fees = await getFeeDetails(props);
    paymentLink = await getPaymentLink(props);

    if (!fees || !paymentLink) {
      errorBoundary(props, 'Invalid initialization');
    }
  } catch (exception) {
    // console.log('error while fetching fee details', exception); // eslint-disable-line
  }

  const list = [
    {
      title: i18n('payment.summary'),
      columns: [
        {
          id: 'description',
          title: i18n('payment.table.description.title'),
        },
        {
          id: 'price',
          title: i18n('payment.table.price.title'),
          align: 'end',
        },
      ],
      items: fees.map((fee: IVariables, id: number) => ({
        id,
        description: locale === 'en' ? fee.feeDescEn : fee.feeDescAr,
        price: fee.FeeAmount,
      })),
    },
  ];

  const totalSection = fees.reduce((total: number, fee: IVariables) => {
    return total + fee.FeeAmount;
  }, 0);

  props.actions.paymentLink.update(paymentLink);
  props.actions.list.update(list);
  props.actions.totalSection.update(totalSection);
};
