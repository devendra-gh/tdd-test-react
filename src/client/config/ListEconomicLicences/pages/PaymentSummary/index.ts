import { IVariables } from '@tamm/app-composer';
import {
  REQUIRES_CUSTOM_LOGIN,
  REQUIRES_UPGRADE_SOP3,
} from 'client/config/utils/customRequireTests';
import { PATH_PAYMENT_SUMMARY } from '../../routes';
import {
  init,
  onPageInit,
  handleTermsConditions,
  startPayment,
} from './functions';

const payment = [
  {
    path: [PATH_PAYMENT_SUMMARY], // path for router
    uniqueId: 'payment', // uniqueId for caching and other purposes
    template: 'paymentSummary', // template name, must be located in index of folder template/index
    title: 'home.title', // title of the page, later it will be read from CMS
    translations: {
      dictionaryId: 'buy-new-car',
      templateId: '4f3b7eee7185484f988ae2d898d666ba',
    },
    init,
    onPageInit,
    props: {
      handleTermsConditions,
      subtitle: 'payment.subtitle',
      description: 'payment.description',
      currentStep: 'applicationSummary',
      stepStatus: {
        selectLicence: 'finish',
      },
      buttons: [
        {
          label: 'button.pay',
          onClick: (props: IVariables) => startPayment(props, true),
          withArrow: true,
        },
        {
          label: 'button.cancel',
          onClick: (props: IVariables) => startPayment(props, false),
          withArrow: true,
          alignIcon: 'start',
          uiType: 'secondary',
        },
      ],

      termsAndConditions: [
        {
          id: 1,
          label: 'terms.label',
        },
      ],
      breadcrumbs: [
        {
          label: 'breadcrumb.level0.title',
          link: 'https://www.tamm.abudhabi/',
        },
        {
          label: 'breadcrumb.level1.title',
          link: 'https://www.tamm.abudhabi/tamm-centers-services',
        },
        {
          label: 'breadcrumb.level2.title',
          link:
            'https://www.tamm.abudhabi/tamm-centers-services/department-of-economic-development',
        },
      ],
    },
    state: {
      mapState: [
        {
          process: (state: IVariables) => ({
            currentStepIndex: state.steps.findIndex(
              ({ name }: IVariables) => name === 'makePayment',
            ),
            title: 'service.process.title',
            steps: state.steps,
          }),
        },
        'user',
        'selectedLicence',
        'businessKey',
        'termsAndConditionsValues',
        'list',
        'totalSection',
        'tags',
        'instanceId',
        'paymentLink',
        'showErrors',
        'loggedIn',
      ],
      mapDispatch: [
        'selectedLicence',
        'instanceId',
        'businessKey',
        'termsAndConditionsValues',
        'list',
        'totalSection',
        'tags',
        'paymentLink',
        'showErrors',
        'startingPayment',
      ],
    },
    requires: [REQUIRES_CUSTOM_LOGIN, REQUIRES_UPGRADE_SOP3],
  },
];

export default payment;
