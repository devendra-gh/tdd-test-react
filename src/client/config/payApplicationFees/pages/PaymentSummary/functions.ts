import { IVariables } from '@tamm/app-composer';
import bpm from 'client/services/bpm';
import { PROCESS_NAME } from 'client/config/payApplicationFees/constants';
import { FORM_STEP_1 } from 'client/config/payApplicationFees/steps';
import { getAnalyticsData } from '../../utils/common';

const onInit = (props: IVariables) => {
  props.actions.stepsStatus.update({
    ...props.stepsStatus,
    [`${FORM_STEP_1}`]: 'finish',
  });
};

const onClick = async (props: IVariables) => {
  getAnalyticsData(
    'pay1',
    { serviceStatus: 'success' },
    { sum: parseFloat(props.totalFees) || props.totalFees },
  );
  await bpm.message(
    PROCESS_NAME,
    {
      businessKey: props.businessKey,
      messageName: 'msgProceedPayment',
      variables: {
        isProceedPayment: true,
      },
    },
    true,
  );
};
const onBack = async (props: IVariables) => {
  await bpm.message(
    PROCESS_NAME,
    {
      businessKey: props.businessKey,
      messageName: 'msgProceedPayment',
      variables: {
        isProceedPayment: false,
      },
    },
    true,
  );
};
export default {
  onInit,
  onClick,
  onBack,
};
