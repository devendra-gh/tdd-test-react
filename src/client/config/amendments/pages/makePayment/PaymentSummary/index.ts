import { REQUIRES_LOGIN } from '@tamm/app-composer/client/constants/requires';
import { noticeTypes } from '@tamm/app-composer/client/constants/types';
import {
  getSteps,
  getStepsStatus,
} from 'client/config/amendments/utils/functions';
import { PROCESS_NAME } from 'client/config/amendments/constants';
import { REQUIRES_SOP3 } from 'client/config/amendments/constants/requires';
import functions from './functions';

const paymentSummary = [
  {
    path: `/amendments/payment-summary`, // path for router
    uniqueId: 'payment-summary', // uniqueId for caching and other purposes
    template: 'summary', // template name, must be located in index of folder template/index
    title: 'title.licenceAmendments', // title of the page, later it will be read from CMS
    props: {
      showSidebar: true,
      currentStep: 'process.makePayment',
      type: noticeTypes.INFO,
      title: 'global.payment',
      content: 'applicationPaymentSummary.content',
      buttons: [],
      pay: functions.pay,
    },
    onPageInit: functions.onPageInit,
    fromProcessState: {
      processName: PROCESS_NAME,
      variables: [
        'apTransactionNumber',
        'submitDate',
        'licenceFees',
        'paymentLink',
      ],
    },
    state: {
      mapState: [
        'loggedIn',
        'businessKey',
        'amendmentServerError',
        {
          steps: getSteps,
          stepsStatus: getStepsStatus('process.makePayment', ''),
        },
      ],
      mapDispatch: ['stepsStatus', 'businessKey', 'amendmentServerError'],
    },
    requires: [REQUIRES_LOGIN, REQUIRES_SOP3],
  },
];

export default paymentSummary;
