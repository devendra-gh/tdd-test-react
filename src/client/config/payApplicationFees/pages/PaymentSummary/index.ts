// import { REQUIRES_LOGIN } from '@tamm/app-composer/client/constants/requires';
import { noticeTypes } from '@tamm/app-composer/client/constants/types';
import { IVariables } from '@tamm/app-composer';
import { PROCESS_NAME } from 'client/config/payApplicationFees/constants';
import { FORM_STEP_2 } from 'client/config/payApplicationFees/steps';
import {
  PATH_PAYMENT_SUMMARY,
  PATH_ERROR,
} from 'client/config/payApplicationFees/routes';
import functions from './functions';

const paymentSummary = [
  {
    path: PATH_PAYMENT_SUMMARY, // path for router
    uniqueId: 'pay-application-fees.payment-summary', // uniqueId for caching and other purposes
    template: 'summary', // template name, must be located in index of folder template/index
    title: 'main.title', // title of the page, later it will be read from CMS
    description: 'notice.economiclicence.pendingPayment.description',
    init: functions.onInit,
    props: {
      currentStep: FORM_STEP_2,
      title: 'notice.economicName.pendingPayment.title',
      subTitle: 'payment',
      description: 'notice.economicName.pendingPayment.text',
      type: noticeTypes.INFO,
      buttons: [
        {
          label: 'common.back',
          onClick: functions.onBack,
          uiType: 'secondary',
          withArrow: true,
          alignIcon: 'start',
        },
        {
          label: 'button.pay',
          onClick: functions.onClick,
          uiType: 'primary',
          withArrow: true,
          alignIcon: 'end',
        },
      ],
    },
    onPageInit: (props: IVariables) => {
      const { feesDetails, totalFees, applicationFeesResult } = props;
      if (applicationFeesResult === 'FEE_API_ERROR') {
        props.history.push(PATH_ERROR);
      }
      const listItems = JSON.parse(feesDetails);
      props.actions.form.update({
        ...props.form,
        feesDetails: listItems,
        totalFees,
      });
      return {
        list: [
          {
            uiType: 'default',
            title: props.i18n('notice.paymentSummary.title'),
            items: listItems,
            columns: [
              {
                id: props.locale === 'en' ? 'feeDescEn' : 'feeDescAr',
                title: props.i18n('label.description'),
              },
              {
                id: 'FeeAmount',
                title: props.i18n('label.fees'),
                align: 'end',
              },
            ],
          },
        ],
        totalSection: props.totalFees,
      };
    },
    fromProcessState: {
      processName: PROCESS_NAME,
      variables: ['feesDetails', 'totalFees', 'applicationFeesResult'],
    },
    state: {
      mapState: [
        'loggedIn',
        'stepsStatus',
        'steps',
        'businessKey',
        'history',
        'form',
      ],
      mapDispatch: ['stepsStatus', 'form'],
    },
    // requires: [REQUIRES_LOGIN],
  },
];

export default paymentSummary;
