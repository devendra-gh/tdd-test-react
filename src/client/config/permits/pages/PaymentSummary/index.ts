import { REQUIRES_LOGIN } from '@tamm/app-composer/client/constants/requires';
import { IVariables } from '@tamm/app-composer';
import { noticeTypes } from '@tamm/app-composer/client/constants/types';
import { PROCESS_NAME, REQUIRES_SOP3 } from 'client/config/permits/constants';
import { getDateFromTimeStamp } from 'client/config/utils/datetime';
import functions from './functions';
import getSteps from '../../utils/getSteps';
import { PATH_PAYMENT_SUMMARY } from '../../utils/constants/pageRoutes';
import checkIfMultiStepPermit from '../../utils/checkIfMultiStepPermit';
import { calculateSum } from '../../utils/common';

const summary = [
  {
    path: `/:serviceName${PATH_PAYMENT_SUMMARY}`, // path for router
    uniqueId: 'payment-summary', // uniqueId for caching and other purposes
    template: 'summary', // template name, must be located in index of folder template/index
    title: 'pageTitle.permitApplication', // title of the page, later it will be read from CMS
    props: {
      type: noticeTypes.INFO,
      title: 'global.payment',
      subTitle: 'global.payment',
      content: 'applicationPaymentSummary.content',
      buttons: [
        {
          label: 'button.pay',
          onClick: functions.onClick(PROCESS_NAME),
          withArrow: true,
        },
      ],
    },
    onPageInit: (props: IVariables) => {
      const { steps } = props;
      const permitsFees = props.permitsFees
        ? JSON.parse(props.permitsFees)
        : [];
      return {
        currentStep: checkIfMultiStepPermit(props)
          ? steps[3].name
          : steps[2].name,
        currentSubStep: checkIfMultiStepPermit(props)
          ? steps[3].subSteps[2]
          : '',
        stepsStatus: functions.getStepStatus(props),
        list: functions.getPaymentSummaryDetails(props),
        totalSection: calculateSum(permitsFees, 'FeeAmount'),
        tags: [
          {
            label: 'notice.refNo',
            value: props.cnNumber,
          },
          {
            label: 'notice.submit',
            value: getDateFromTimeStamp(props.submitDate),
          },
        ],
      };
    },
    fromProcessState: {
      processName: PROCESS_NAME,
      variables: ['permitsFees', 'paymentLink', 'submitDate'],
    },

    state: {
      mapState: [
        'loggedIn',
        'stepsStatus',
        {
          steps: (state: IVariables) => {
            return getSteps(state);
          },
        },
        'serviceType',
        'businessKey',
        'history',
        'permitType',
        'permitInfo',
        'companyDetails',
        'user',
      ],
      mapDispatch: ['stepsStatus'],
    },
    requires: [REQUIRES_LOGIN, REQUIRES_SOP3],
  },
];

export default summary;
