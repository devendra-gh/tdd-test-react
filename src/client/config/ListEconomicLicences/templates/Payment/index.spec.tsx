import React from 'react';
import { shallow } from 'enzyme';
import PaymentSummary from './index';

jest.mock('@tamm/app-composer', () => ({
  withTemplateHooks: (component: any) => component,
}));

// eslint-disable-next-line no-console
console.log = () => {};
console.info = () => {};
console.warn = () => {};
console.error = () => {};

describe('Summary Template', () => {
  let props: any;
  beforeEach(() => {
    props = {
      i18n: jest.fn(),
      locale: 'en',
      title: 'Notice',
      status: 'success',
      handleTermsConditions: jest.fn(),
      subtitle: 'Test subtitle',
      description: 'Payment Template',
      termsAndConditionsValues: [true],
      termsAndConditions: {},
      breadcrumbs: [],
      totalSection: 200,
      businessKey: '123456',
      instanceId: '123456',
      showErrors: false,
      startingPayment: false,
      process: {
        title: 'process title',
        steps: [
          {
            name: 'process step',
          },
        ],
      },
      stepStatus: { 'process step': 'finish' },
      buttons: [
        {
          label: 'button',
          onClick: jest.fn(),
          uiType: 'primary',
          withArraow: false,
        },
      ],
      list: [
        {
          labelHeading: 'test label heading',
          valueHeading: 'test value heading',
          listDetails: {
            label: 'test label',
            value: 'test value',
          },
        },
      ],
    };
  });

  it('should successfully render the summary template', () => {
    const component = shallow(<PaymentSummary {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('should successfully render the summary template with spinner when starting payment', () => {
    props.startingPayment = true;
    const component = shallow(<PaymentSummary {...props} />);
    expect(component).toMatchSnapshot();
  });
});
