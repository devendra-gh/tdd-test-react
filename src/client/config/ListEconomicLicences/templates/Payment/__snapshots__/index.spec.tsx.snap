// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary Template should successfully render the summary template 1`] = `
<Container
  locale="en"
  sidebar={
    <SidebarTemplate
      eligibilityCard={null}
      label=""
      lists={Array []}
      process={
        Object {
          "steps": Array [
            Object {
              "label": undefined,
              "name": "process step",
              "status": "finish",
            },
          ],
          "title": undefined,
        }
      }
      relatedJourney={null}
      relevantEntity={
        Object {
          "entities": Array [
            Object {
              "address": undefined,
              "email": "contact.abudhabi.ae",
              "id": 0,
              "logo": "/images/DED-logo.png",
              "officeHours": Object {
                "status": "default",
                "workingHours": Object {
                  "friday": Object {
                    "closed": false,
                    "end": "10 pm",
                    "label": "5 am - 10 pm",
                    "start": "5 am",
                  },
                  "monday": Object {
                    "closed": false,
                    "end": "10 pm",
                    "label": "5 am - 10 pm",
                    "start": "5 am",
                  },
                  "saturday": Object {
                    "closed": true,
                  },
                  "sunday": Object {
                    "closed": false,
                    "end": "10 pm",
                    "label": "5 am - 10 pm",
                    "start": "5 am",
                  },
                  "thursday": Object {
                    "closed": false,
                    "end": "10 pm",
                    "label": "5 am - 10 pm",
                    "start": "5 am",
                  },
                  "tuesday": Object {
                    "closed": false,
                    "end": "10 pm",
                    "label": "5 am - 10 pm",
                    "start": "5 am",
                  },
                  "wednesday": Object {
                    "closed": false,
                    "end": "10 pm",
                    "label": "5 am - 10 pm",
                    "start": "5 am",
                  },
                },
              },
              "phones": Array [
                "+971 2 815 8888",
              ],
              "publicServiceHours": Object {
                "status": "24/7",
              },
              "website": "www.adeconomy.ae",
            },
          ],
          "title": undefined,
        }
      }
      relevantEntityLink={
        Object {
          "label": undefined,
        }
      }
      socialSharing={null}
    />
  }
>
  <div
    className=""
  >
    <h1
      className="syb-title"
      style={
        Object {
          "marginBottom": "2rem",
        }
      }
    />
    <p
      style={
        Object {
          "color": "#000",
        }
      }
    />
  </div>
  <div
    className="summary"
  >
    <LocalizedComponent(Form)
      backButton={null}
      breadcrumbs={Array []}
      businessKey="123456"
      buttons={
        Array [
          Object {
            "label": "button",
            "onClick": [MockFunction],
            "uiType": "primary",
            "withArraow": false,
          },
        ]
      }
      cancelLink={
        Object {
          "href": "",
          "label": undefined,
          "onClick": [Function],
        }
      }
      description="Payment Template"
      handleTermsConditions={[MockFunction]}
      help=""
      i18n={
        [MockFunction] {
          "calls": Array [
            Array [
              "departmentOfEconomicDevelopment.address",
            ],
            Array [
              "sidebar.home.label",
            ],
            Array [
              "sidebar.home.relevantEntityLink",
            ],
            Array [
              "process title",
            ],
            Array [
              "process step",
            ],
            Array [
              "relevant_entity",
            ],
            Array [
              "Test subtitle",
            ],
            Array [
              "Payment Template",
            ],
            Array [
              "button",
            ],
            Array [
              undefined,
            ],
            Array [
              "global.aed",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      id=""
      instanceId="123456"
      list={
        Array [
          Object {
            "labelHeading": "test label heading",
            "listDetails": Object {
              "label": "test label",
              "value": "test value",
            },
            "valueHeading": "test value heading",
          },
        ]
      }
      locale="en"
      process={
        Object {
          "steps": Array [
            Object {
              "name": "process step",
            },
          ],
          "title": "process title",
        }
      }
      showErrors={false}
      startingPayment={false}
      status="success"
      stepStatus={
        Object {
          "process step": "finish",
        }
      }
      submitButton={
        Object {
          "label": undefined,
          "onClick": [Function],
          "withArrow": true,
        }
      }
      subtitle="Test subtitle"
      termsAndConditions={Object {}}
      termsAndConditionsValues={
        Array [
          true,
        ]
      }
      title="Notice"
      totalSection={200}
      validateStatus={null}
    >
      <div
        className="summary__group summary__group__table"
      >
        <LocalizedComponent(Table)
          clickable={false}
          commonSelection="unchecked"
          currPage={null}
          disabledSelectionVisible={false}
          editable={false}
          filter={null}
          headerHidden={false}
          i18n={
            [MockFunction] {
              "calls": Array [
                Array [
                  "departmentOfEconomicDevelopment.address",
                ],
                Array [
                  "sidebar.home.label",
                ],
                Array [
                  "sidebar.home.relevantEntityLink",
                ],
                Array [
                  "process title",
                ],
                Array [
                  "process step",
                ],
                Array [
                  "relevant_entity",
                ],
                Array [
                  "Test subtitle",
                ],
                Array [
                  "Payment Template",
                ],
                Array [
                  "button",
                ],
                Array [
                  undefined,
                ],
                Array [
                  "global.aed",
                ],
              ],
              "results": Array [
                Object {
                  "type": "return",
                  "value": undefined,
                },
                Object {
                  "type": "return",
                  "value": undefined,
                },
                Object {
                  "type": "return",
                  "value": undefined,
                },
                Object {
                  "type": "return",
                  "value": undefined,
                },
                Object {
                  "type": "return",
                  "value": undefined,
                },
                Object {
                  "type": "return",
                  "value": undefined,
                },
                Object {
                  "type": "return",
                  "value": undefined,
                },
                Object {
                  "type": "return",
                  "value": undefined,
                },
                Object {
                  "type": "return",
                  "value": undefined,
                },
                Object {
                  "type": "return",
                  "value": undefined,
                },
                Object {
                  "type": "return",
                  "value": undefined,
                },
              ],
            }
          }
          isSingleSelect={false}
          items={Array []}
          labelHeading="test label heading"
          listDetails={
            Object {
              "label": "test label",
              "value": "test value",
            }
          }
          multiActions={null}
          onActionClick={[Function]}
          onClick={[Function]}
          onFilter={[Function]}
          onPageResize={[Function]}
          onPageTurn={[Function]}
          onRefreshClick={[Function]}
          onSearch={[Function]}
          onSelectionChange={[Function]}
          onSort={[Function]}
          onToggle={[Function]}
          pageResizeOptions={Array []}
          pageSize={null}
          rowVerticalAlign="top"
          search=""
          searchable={false}
          selectable={false}
          singleActions={null}
          size="default"
          sort={null}
          status=""
          title=""
          total={null}
          valueHeading="test value heading"
        />
      </div>
      <div
        className="summary__group"
      >
        <LocalizedComponent(Total)
          buttonDisabled={false}
          i18n={[Function]}
          isButtonVisible={false}
          isValueFirst={false}
          onClick={null}
          unit=""
          value={200}
        />
      </div>
      <div
        className="summary__group"
      >
        <TermsAndConditions
          checked={
            Array [
              true,
            ]
          }
          handleClick={[Function]}
          i18n={
            [MockFunction] {
              "calls": Array [
                Array [
                  "departmentOfEconomicDevelopment.address",
                ],
                Array [
                  "sidebar.home.label",
                ],
                Array [
                  "sidebar.home.relevantEntityLink",
                ],
                Array [
                  "process title",
                ],
                Array [
                  "process step",
                ],
                Array [
                  "relevant_entity",
                ],
                Array [
                  "Test subtitle",
                ],
                Array [
                  "Payment Template",
                ],
                Array [
                  "button",
                ],
                Array [
                  undefined,
                ],
                Array [
                  "global.aed",
                ],
              ],
              "results": Array [
                Object {
                  "type": "return",
                  "value": undefined,
                },
                Object {
                  "type": "return",
                  "value": undefined,
                },
                Object {
                  "type": "return",
                  "value": undefined,
                },
                Object {
                  "type": "return",
                  "value": undefined,
                },
                Object {
                  "type": "return",
                  "value": undefined,
                },
                Object {
                  "type": "return",
                  "value": undefined,
                },
                Object {
                  "type": "return",
                  "value": undefined,
                },
                Object {
                  "type": "return",
                  "value": undefined,
                },
                Object {
                  "type": "return",
                  "value": undefined,
                },
                Object {
                  "type": "return",
                  "value": undefined,
                },
                Object {
                  "type": "return",
                  "value": undefined,
                },
              ],
            }
          }
          items={Object {}}
          showErrors={false}
        />
      </div>
    </LocalizedComponent(Form)>
  </div>
</Container>
`;

exports[`Summary Template should successfully render the summary template with spinner when starting payment 1`] = `
<Container
  locale="en"
  sidebar={
    <SidebarTemplate
      eligibilityCard={null}
      label=""
      lists={Array []}
      process={
        Object {
          "steps": Array [
            Object {
              "label": undefined,
              "name": "process step",
              "status": "finish",
            },
          ],
          "title": undefined,
        }
      }
      relatedJourney={null}
      relevantEntity={
        Object {
          "entities": Array [
            Object {
              "address": undefined,
              "email": "contact.abudhabi.ae",
              "id": 0,
              "logo": "/images/DED-logo.png",
              "officeHours": Object {
                "status": "default",
                "workingHours": Object {
                  "friday": Object {
                    "closed": false,
                    "end": "10 pm",
                    "label": "5 am - 10 pm",
                    "start": "5 am",
                  },
                  "monday": Object {
                    "closed": false,
                    "end": "10 pm",
                    "label": "5 am - 10 pm",
                    "start": "5 am",
                  },
                  "saturday": Object {
                    "closed": true,
                  },
                  "sunday": Object {
                    "closed": false,
                    "end": "10 pm",
                    "label": "5 am - 10 pm",
                    "start": "5 am",
                  },
                  "thursday": Object {
                    "closed": false,
                    "end": "10 pm",
                    "label": "5 am - 10 pm",
                    "start": "5 am",
                  },
                  "tuesday": Object {
                    "closed": false,
                    "end": "10 pm",
                    "label": "5 am - 10 pm",
                    "start": "5 am",
                  },
                  "wednesday": Object {
                    "closed": false,
                    "end": "10 pm",
                    "label": "5 am - 10 pm",
                    "start": "5 am",
                  },
                },
              },
              "phones": Array [
                "+971 2 815 8888",
              ],
              "publicServiceHours": Object {
                "status": "24/7",
              },
              "website": "www.adeconomy.ae",
            },
          ],
          "title": undefined,
        }
      }
      relevantEntityLink={
        Object {
          "label": undefined,
        }
      }
      socialSharing={null}
    />
  }
>
  <Component />
  <div
    className=""
  >
    <h1
      className="syb-title"
      style={
        Object {
          "marginBottom": "2rem",
        }
      }
    />
    <p
      style={
        Object {
          "color": "#000",
        }
      }
    />
  </div>
  <div
    className="summary"
  >
    <LocalizedComponent(Form)
      backButton={null}
      breadcrumbs={Array []}
      businessKey="123456"
      buttons={
        Array [
          Object {
            "label": "button",
            "onClick": [MockFunction],
            "uiType": "primary",
            "withArraow": false,
          },
        ]
      }
      cancelLink={
        Object {
          "href": "",
          "label": undefined,
          "onClick": [Function],
        }
      }
      description="Payment Template"
      handleTermsConditions={[MockFunction]}
      help=""
      i18n={
        [MockFunction] {
          "calls": Array [
            Array [
              "departmentOfEconomicDevelopment.address",
            ],
            Array [
              "sidebar.home.label",
            ],
            Array [
              "sidebar.home.relevantEntityLink",
            ],
            Array [
              "process title",
            ],
            Array [
              "process step",
            ],
            Array [
              "relevant_entity",
            ],
            Array [
              "Test subtitle",
            ],
            Array [
              "Payment Template",
            ],
            Array [
              "button",
            ],
            Array [
              undefined,
            ],
            Array [
              "global.aed",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      id=""
      instanceId="123456"
      list={
        Array [
          Object {
            "labelHeading": "test label heading",
            "listDetails": Object {
              "label": "test label",
              "value": "test value",
            },
            "valueHeading": "test value heading",
          },
        ]
      }
      locale="en"
      process={
        Object {
          "steps": Array [
            Object {
              "name": "process step",
            },
          ],
          "title": "process title",
        }
      }
      showErrors={false}
      startingPayment={true}
      status="success"
      stepStatus={
        Object {
          "process step": "finish",
        }
      }
      submitButton={
        Object {
          "label": undefined,
          "onClick": [Function],
          "withArrow": true,
        }
      }
      subtitle="Test subtitle"
      termsAndConditions={Object {}}
      termsAndConditionsValues={
        Array [
          true,
        ]
      }
      title="Notice"
      totalSection={200}
      validateStatus={null}
    >
      <div
        className="summary__group summary__group__table"
      >
        <LocalizedComponent(Table)
          clickable={false}
          commonSelection="unchecked"
          currPage={null}
          disabledSelectionVisible={false}
          editable={false}
          filter={null}
          headerHidden={false}
          i18n={
            [MockFunction] {
              "calls": Array [
                Array [
                  "departmentOfEconomicDevelopment.address",
                ],
                Array [
                  "sidebar.home.label",
                ],
                Array [
                  "sidebar.home.relevantEntityLink",
                ],
                Array [
                  "process title",
                ],
                Array [
                  "process step",
                ],
                Array [
                  "relevant_entity",
                ],
                Array [
                  "Test subtitle",
                ],
                Array [
                  "Payment Template",
                ],
                Array [
                  "button",
                ],
                Array [
                  undefined,
                ],
                Array [
                  "global.aed",
                ],
              ],
              "results": Array [
                Object {
                  "type": "return",
                  "value": undefined,
                },
                Object {
                  "type": "return",
                  "value": undefined,
                },
                Object {
                  "type": "return",
                  "value": undefined,
                },
                Object {
                  "type": "return",
                  "value": undefined,
                },
                Object {
                  "type": "return",
                  "value": undefined,
                },
                Object {
                  "type": "return",
                  "value": undefined,
                },
                Object {
                  "type": "return",
                  "value": undefined,
                },
                Object {
                  "type": "return",
                  "value": undefined,
                },
                Object {
                  "type": "return",
                  "value": undefined,
                },
                Object {
                  "type": "return",
                  "value": undefined,
                },
                Object {
                  "type": "return",
                  "value": undefined,
                },
              ],
            }
          }
          isSingleSelect={false}
          items={Array []}
          labelHeading="test label heading"
          listDetails={
            Object {
              "label": "test label",
              "value": "test value",
            }
          }
          multiActions={null}
          onActionClick={[Function]}
          onClick={[Function]}
          onFilter={[Function]}
          onPageResize={[Function]}
          onPageTurn={[Function]}
          onRefreshClick={[Function]}
          onSearch={[Function]}
          onSelectionChange={[Function]}
          onSort={[Function]}
          onToggle={[Function]}
          pageResizeOptions={Array []}
          pageSize={null}
          rowVerticalAlign="top"
          search=""
          searchable={false}
          selectable={false}
          singleActions={null}
          size="default"
          sort={null}
          status=""
          title=""
          total={null}
          valueHeading="test value heading"
        />
      </div>
      <div
        className="summary__group"
      >
        <LocalizedComponent(Total)
          buttonDisabled={false}
          i18n={[Function]}
          isButtonVisible={false}
          isValueFirst={false}
          onClick={null}
          unit=""
          value={200}
        />
      </div>
      <div
        className="summary__group"
      >
        <TermsAndConditions
          checked={
            Array [
              true,
            ]
          }
          handleClick={[Function]}
          i18n={
            [MockFunction] {
              "calls": Array [
                Array [
                  "departmentOfEconomicDevelopment.address",
                ],
                Array [
                  "sidebar.home.label",
                ],
                Array [
                  "sidebar.home.relevantEntityLink",
                ],
                Array [
                  "process title",
                ],
                Array [
                  "process step",
                ],
                Array [
                  "relevant_entity",
                ],
                Array [
                  "Test subtitle",
                ],
                Array [
                  "Payment Template",
                ],
                Array [
                  "button",
                ],
                Array [
                  undefined,
                ],
                Array [
                  "global.aed",
                ],
              ],
              "results": Array [
                Object {
                  "type": "return",
                  "value": undefined,
                },
                Object {
                  "type": "return",
                  "value": undefined,
                },
                Object {
                  "type": "return",
                  "value": undefined,
                },
                Object {
                  "type": "return",
                  "value": undefined,
                },
                Object {
                  "type": "return",
                  "value": undefined,
                },
                Object {
                  "type": "return",
                  "value": undefined,
                },
                Object {
                  "type": "return",
                  "value": undefined,
                },
                Object {
                  "type": "return",
                  "value": undefined,
                },
                Object {
                  "type": "return",
                  "value": undefined,
                },
                Object {
                  "type": "return",
                  "value": undefined,
                },
                Object {
                  "type": "return",
                  "value": undefined,
                },
              ],
            }
          }
          items={Object {}}
          showErrors={false}
        />
      </div>
    </LocalizedComponent(Form)>
  </div>
</Container>
`;
